d2common/d2fileformats/d2ds1/ds1_layers.go

```diff
@@ -0,0 +1,432 @@
+package d2ds1
+
+const (
+	maxWallLayers         = 4
+	maxOrientationLayers  = 4
+	maxFloorLayers        = 2
+	maxShadowLayers       = 1
+	maxSubstitutionLayers = 1
+)
+
+type layerGroupType int
+
+const (
+	floorLayerGroup layerGroupType = iota
+	wallLayerGroup
+	orientationLayerGroup
+	shadowLayerGroup
+	substitutionLayerGroup
+)
+
+type layerGroup []*layer
+
+type ds1Layers struct {
+	width, height int
+	Floors        layerGroup
+	Walls         layerGroup
+	Orientations  layerGroup
```

Comment:
why did you splited it? IMO Wall and orientation should be one

Reply:
yeah, you're right

Reply:
fixed in 41c1d8e